### ISSUE TITLE
Save and continue: where users are redirected to after saving/publishing

### ISSUE THREAD

**Issue by mattots** [wagtail/wagtail#694]
There may be good reasons behind the decision to have this work the way it does... but to my mind it would make a lot more sense if the '+' button on the Wagtail userbar added a CHILD page of the currently viewed page. After all, in the CMS a user selects the PARENT page then clicks a button to add a child page, so it seems odd on the front end to have to go to a page at the SAME level as the page you want to create. And if there isn't a page already at that level then there's no way to add it via the front end at all.

The particular scenario that prompted this thought was in setting up a blog - with a BlogIndexPage model and BlogPostPage model. For the user wanting to add a blog post from the front end (using the userbar) they would have to first go to an existing blog post before adding a new one, whereas the logical thing to do would be to go to the blog index page and add a new post from there.

The logic / typical through process is the same for any kind of pages really though.

**Comment by mattots** [comment]
It looks like this has been resolved now in #728, so I don't know if this is the right place to add this or not but here goes...

When dropping into the admin interface from the front end via the userbar, it seems to me that it would make more sense if after you publish/submit your changes you are returned to the front end page on which you started. The point of the userbar, surely is to allow someone to quickly drop in and out of editing mode, so it seems to make sense to return them to the page they started on.

**Comment by davecranwell** [comment]
That's perhaps how _you_ most use the userbar, but I think it would be a mistake to make that the behaviour by default. 

Let's avoid the honeypot of telling other people what the "correct" way of using Wagtail is. Instead I'd suggest that how the userbar functions should be down to individual user settings, edited from their account.

So you'd have settings for 
1) whether published a page accessed from the userbar returns you to the front end
2) whether pressing "publish" at *any* time returns you to the Explorer, or leaves you in the page editor
3) what order pages in the explorer should be shown in by default
 ... more

I say give users the power to choose what happens next.

**Comment by mattots** [comment]
@davecranwell Fair point - I'm all for more user choice, so long as it doesn't over complicate things or end up as a setting that few people will ever notice or bother with.

I guess my assumption is just based on the idea of the userbar as a way of quickly "dipping in and out" of the admin interface while primarily using the front end. But I fully appreciate there are other scenarios where this doesn't hold true.

**Comment by mattots** [comment]
I guess another option might be to open the admin interface in a new window/tab, kind of mirroring the 'Preview' button functionality in the admin.

**Comment by thibaudcolas** [comment]
I’d like to co-opt this long-standing ticket to say we shouldn’t make this configurable. Instead, saving/publishing a page should always take the user back to the editing form for said page. If the user wants to go back to a different page, they can do so with the "back" button in their browser, or open the editing form in a different tab.

My rationale is – above all else, "where you go when submitting a form" needs to be consistent and predictable. With this in mind, it shouldn’t be configurable, nor vary from scenario to scenario. Taking the user back to the form will also be the most consistent with our expected future auto-save experience ([roadmap: Autosave #24](https://github.com/wagtail/roadmap/issues/24)).

cc @vossisboss who raised this

**Comment by BigWhale** [comment]
I would just like to add that our first batch of alpha testers that tried Wagtail immediately reported that closing the editor after saving/publishing of a page is one of the biggest deal breakers and a big usability problem in day-to-day work.

We are working on a CMS for many different news or news-like sites and this is something that we will have to handle somehow. Our use case is that users tend to click save as draft quite often in fear of losing their work, and when they are writing an article on hot topic, they want to publish things ASAP so they write a short article just to get the news out and then they proceed to write a longer version.

**Comment by gasman** [comment]
Thanks for the feedback @BigWhale! I suspect - albeit without any hard evidence - that the pattern of publishing a page and then working on it some more is the exception rather than the rule, and that for most editors, publishing (or submitting for moderation) is usually the final action of working with a given page, and being taken out of that editing workflow to select a new item is more natural. Of course, it's hard to gather data on that here, since the people for whom the current behaviour works fine are less likely to be seeking out this Github issue :-)

I wonder if this is something that makes sense to configure on a per-page-model basis, if there's an "article" page type where the "publish then continue editing" pattern is more typical - although I realise that's at odds with @thibaudcolas's desire for it not to vary from scenario to scenario. I think there's general acceptance that we shouldn't leave this decision up to the editor (either through a user preference or a django-admin-style set of "publish" / "publish and continue editing" buttons) as it's a micro-management detail that users shouldn't really have to think about - but letting it be a micro-management detail for the _developer_ to determine seems a bit more palatable.

**Comment by thibaudcolas** [comment]
We’ve decided to explicitly mark this as "Needs user testing" rather than "needs design decision", and review users’ expectations the next time we have formal testing.

In the meantime if anyone wants to share their feedback as CMS users here it’s always helpful.

**Comment by lb-** [comment]
Here's a related issue with the way the userbar loads the page for the 'show in explorer' button. Not the same problem but worth reviewing in the context of changes planned. 

https://github.com/wagtail/wagtail/issues/4105

**Comment by smark-1** [comment]
I wonder if ModelViewSet/Snippets edit view should also take the user back to the edit form. I personally would prefer this behavior for ViewSet and Pages. Also if Pages are updated to go back to the edit form doing the same for ModelViewSet and Snippets would be good for consistency.

**Comment by laymonage** [comment]
FYI snippets already do that if they have DraftStateMixin enabled, so the behaviour is currently consistent.

**Comment by smark-1** [comment]
Is there a way to do that for ModelViewSet as well?
